### ISSUE TITLE
Caffe : Segmentation fault

### ISSUE THREAD

**Issue by larenzhang** [saiprabhakar/DeepDriving#3]
I have executed the command： make all ,make test and there are no errors when I install caffe on ubuntu 16.04(cuda 8.0+Nvida Titan X).However,it occurs "Segmentation fault" when I excute "make runtest"  as the follwing:
Makefile:412: recipe for target 'runtest' failed
make: *** [runtest] Segmentation fault (core dumped)

and when I train the model with the dataset downloading from the website:http://deepdriving.cs.princeton.edu/ ,there are also errors:
I1006 16:50:11.402531 25092 net.cpp:84] Creating Layer data
I1006 16:50:11.402541 25092 net.cpp:373] data -> data
I1006 16:50:11.402570 25092 net.cpp:373] data -> label
I1006 16:50:11.402595 25092 net.cpp:113] Setting up data
Segmentation fault (core dumped)

Do you have the same experience?

**Comment by saiprabhakar** [comment]
I am not what, but I think this is one of the common Caffe installation errors. You should be able to solve by referring BVLC/caffe.

**Comment by larenzhang** [comment]
Thanks for your respond.Maybe the version of caffe is too old ,so that it's hard to compatible with my development enviroment.Therefore,I have installed the newest version of caffe and it make runtest well.Even though I can't share memory with torcs,it's not so necessary.I just want to train the AlexNet model with the given dataset(TORCS_TRAIN_IF,leveldb dataset).Sadly,it still occurs a question as follows:
F1012 12:17:12.502512 15283 euclidean_loss_layer.cpp:12] Check failed: bottom[0]->count(1) == bottom[1]->count(1) (14 vs. 1) Inputs must have the same dimension.
*** Check failure stack trace: ***
Aborted (core dumped)

It seems that the dimension of the neural network's output is 14,but the dimenison of the label is just 1.I don't have the original dataset and it's encoded to leveldb,so that I can't validate what the detailed structure of the data especially the label'.

Have you ever train the model with the given dataset from  the website:http://deepdriving.cs.princeton.edu/ successfully?

**Comment by Azpril45** [comment]
Dear @larenzhang @saiprabhakar  , thank you for sharing this repo. 

I got a problem when tried to train a new model. 
I1102 13:26:38.987882 14212 net.cpp:84] Creating Layer data 
I1102 13:26:38.987896 14212 net.cpp:373] data -> data 
I1102 13:26:38.987915 14212 net.cpp:373] data -> label 
I1102 13:26:38.987929 14212 net.cpp:113] Setting up data 
Segmentation fault (core dumped) 

I have downloaded the training data, and replace it in pre_trained.  Besides, I have already followed the instructions and changed the data  layer cpp. It still happened. 
Is that because of the version of Caffe like you said above?
I got no problem with the test. It goes well. Can you help me figure out what is wrong here? 

Thank you for your patience. 
Have a nice day! 

Best regards, 
Tony
--
